### DISCUSSION TITLE
Serial Number in DN

### DISCUSSION THREAD

**Discussion by talcher** [Keyfactor/ejbca-ce#612] [Q&A] [answered]
Hello the EJBCA team,

in an end entity profile, there is the opportunity to add the field "serialNumber, Serial number (in DN)". Correct me if I am wrong, but it seems to me that this value shall be provided to EJBCA upon creating a new certificate.
Is there a way to automatically populate this field with the serial number given to the certificate by the CA?

Thank you for your help!

Alcher

**Comment by primetomas** [comment]
The SerialNumber DN components is not the same as certificate serial number. These are different fields. The SerialNumber in DN is typically for device serial numbers, i.e. if you manufacture phones, routers, laptops, telco base stations, etc you identify these devices by a serial number. This is the serial number you put in the subject DN.

the certificate serial number is not part of the subject DN, and is always populated automatically. 

See [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280) for technical details.

**Reply by talcher** [reply]
Thank you, Tomas, for your answer.

I knew that these two fields are different and that the certificate serial number is populated automatically.
My question was: is there a way to populate the field of the DN with the value of the automatically-generated certificate serial number?
I guess the answer will be no. I am asking just in case the answer were yes!

**Reply by primetomas** [reply, marked as answer]
Correct, the answer is no. Why would you like these two fields to be the same? Duplicating information is rarely a good thing.

**Reply by talcher** [reply]
Thank you for your help, Tomas.
Take care!
